modules/backendscheduler/backendscheduler.go

```diff
@@ -256,7 +259,7 @@ func (s *BackendScheduler) Next(ctx context.Context, req *tempopb.NextJobRequest
 		level.Info(log.Logger).Log("msg", "assigned job to worker", "job_id", j.ID, "worker", req.WorkerId)
 
 		return resp, nil
-	default:
+	case <-timeoutCtx.Done():
```

Comment:
This dramatically lowered the instances of "no jobs found" results and increased compaction throughput.

Reply:
noted in my review, but this was a miss from before. good catch!